src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -420,25 +420,24 @@ InvokeHostFunctionOpFrame::doApply(Application& app, AbstractLedgerTxn& ltx,
                     TTLEntry = ttlLtxe.current().data.ttl();
                 }
 
-                if (shouldAddEntry)
+                auto leBuf = toCxxBuf(le);
+
+                // For entry types that don't have an TTLEntry (i.e.
+                // Accounts), the rust host expects an "empty" CxxBuf such
+                // that the buffer has a non-null pointer that points to an
+                // empty byte vector
+                auto ttlBuf =
+                    TTLEntry ? toCxxBuf(*TTLEntry)
+                             : CxxBuf{std::make_unique<std::vector<uint8_t>>()};
+
+                entrySize = static_cast<uint32>(leBuf.data->size());
+                if (TTLEntry)
                 {
-                    auto leBuf = toCxxBuf(le);
-
-                    // For entry types that don't have an TTLEntry (i.e.
-                    // Accounts), the rust host expects an "empty" CxxBuf such
-                    // that the buffer has a non-null pointer that points to an
-                    // empty byte vector
-                    auto ttlBuf =
-                        TTLEntry
-                            ? toCxxBuf(*TTLEntry)
-                            : CxxBuf{std::make_unique<std::vector<uint8_t>>()};
-
-                    entrySize = static_cast<uint32>(leBuf.data->size());
-                    if (TTLEntry)
-                    {
-                        entrySize += static_cast<uint32_t>(ttlBuf.data->size());
-                    }
+                    entrySize += static_cast<uint32_t>(ttlBuf.data->size());
```

Comment:
I haven't noticed this code before - why do we increase the entry size here? TTL entries shouldn't be observable in non-refundable resources. Using it for one particular limit is rather misleading.

Reply:
Hmm that's a good point. TTL write fees come out of refundable fee, but it looks like all Soroban ops increase the read bytes when reading TTL entries. Should we be charging from refundable fee instead for the reads? That sounds more complex with the way things are set up now. Maybe we just don't charge for TTL reads? That isn't ideal though.

Reply:
Actually, putting this into refundable fee won't be correct either because we perform reads no matter what.  TBH I'm not sure if the confusion potential caused by this is worth the minor benefit this provides. TTL entries are constant size, so they can be thought of as a constant factor added to the entry read cost. Of course, not every entry has TTL entry, but it's seems ok to just charge the 'upper bound' cost, given that majority of entries do have TTL.

Reply:
Personally I'm fine with the current approach. Ideally, we'd charge TTL writes via writeBytes. We can't do this though because we don't know if we will perform a TTL write until runtime. For reads this isn't the case, since we know we will perform all reads. I don't like overcharging for entries that don't have a TTL entry either. To the end user I think the interface makes sense. When you update the TTL, you get charged from the refundableFee. When you read a Soroban entry, there's a little more read bytes overhead than a Stellar classic entry. While there is an inconsistency in the implementation details, it's pretty deep in the stack so I doubt it would be a huge issue for developers.

Reply:
>  Ideally, we'd charge TTL writes via writeBytes. 

No, this is not what we have discussed so far. TTL entries should be transparent to the users and we don't want to have them transparent only in some contexts (like footprint), but not the others (like RW bytes). We also do charge the writes as a part of refundable fee.

> While there is an inconsistency in the implementation details, it's pretty deep in the stack so I doubt it would be a huge issue for developers.

This is an issue for preflight/simulation. It brings this implementation detail up to the layers that shouldn't really worry about that.

> I don't like overcharging for entries that don't have a TTL entry either. 

It's not like we are really overcharging anything, it's just that reading *any* entry costs the same fee. If we have issues due to number of ledger entries read, then we can increase the entry read fees and/or reduce the tx/ledger limits on the number of entry reads. Adding a few stroops to the fee and a few bytes to the read bytes limit will not in any way prevent any issues. It will however create annoying issues for any downstream customer that tries to fill the soroban resources.

Reply:
Yeah I agree that adding TTL bytes to the read counter is messy. We don't expose these entries anywhere, so expecting them to be accounted for as bytes read is confusing.

I updated the PR to not count TTL readBytes in InvokeHostFunctionOp, and only load soroban entries that aren't expired. If we're all in agreement with this approach, I'll update the other ops and increase the entry read cost (let me know if I misunderstood this discussion).

Reply:
Thanks - let's also update other ops in this PR then to have consistent semantics.

Reply:
Updated for all ops